Sources/Workspace/Workspace.swift

```diff
@@ -290,9 +287,18 @@ public class Workspace {
     }
 
     /// Create the cache directories.
-    private func createCacheDirectories() throws {
-        try fileSystem.createDirectory(repositoryManager.path, recursive: true)
-        try fileSystem.createDirectory(checkoutsPath, recursive: true)
+    private func createCacheDirectories(engine: DiagnosticsEngine) {
+        do {
+            try fileSystem.createDirectory(repositoryManager.path, recursive: true)
+        } catch {
+            engine.emit(error)
+        }
+        
+        do {
+            try fileSystem.createDirectory(checkoutsPath, recursive: true)
+        } catch {
+            engine.emit(error)
+        }
```

Comment:
I don't think we need separate do catch here, a failure here would mean this method needs to be retried anyway.